src/EditorFeatures/TestUtilities/Diagnostics/AbstractUserDiagnosticTest.cs

```diff
@@ -199,27 +199,9 @@ protected Document GetDocumentAndAnnotatedSpan(TestWorkspace workspace, out stri
             }
 
             var diagnostic = diagnostics.First();
-            Func<Document, ImmutableHashSet<string>, CancellationToken, Task<IEnumerable<Diagnostic>>> getDocumentDiagnosticsAsync =
-                async (d, diagIds, c) =>
-                {
-                    var root = await d.GetSyntaxRootAsync();
-                    var diags = await testDriver.GetDocumentDiagnosticsAsync(provider, d, root.FullSpan);
-                    diags = diags.Where(diag => diagIds.Contains(diag.Id));
-                    return diags;
-                };
-
-            Func<Project, bool, ImmutableHashSet<string>, CancellationToken, Task<IEnumerable<Diagnostic>>> getProjectDiagnosticsAsync =
-                async (p, includeAllDocumentDiagnostics, diagIds, c) =>
-                {
-                    var diags = includeAllDocumentDiagnostics
-                        ? await testDriver.GetAllDiagnosticsAsync(provider, p)
-                        : await testDriver.GetProjectDiagnosticsAsync(provider, p);
-                    diags = diags.Where(diag => diagIds.Contains(diag.Id));
-                    return diags;
-                };
-
             var diagnosticIds = ImmutableHashSet.Create(diagnostic.Id);
-            var fixAllDiagnosticProvider = new FixAllState.FixAllDiagnosticProvider(diagnosticIds, getDocumentDiagnosticsAsync, getProjectDiagnosticsAsync);
+            var fixAllDiagnosticProvider = new FixAllDiagnosticProvider(provider, testDriver, diagnosticIds);
```

Comment:
Its been a long time since I looked at this code, so I can't recall the reason why the code is structured this way. However, I would recommend trying to get rid of [TestDiagnosticAnalyzerDriver](http://source.roslyn.io/#Roslyn.Services.Test.Utilities/Diagnostics/TestDiagnosticAnalyzerDriver.cs,6b06ccfaceb45a50,references) and [TestDiagnosticAnalyzerService](http://source.roslyn.io/#Roslyn.Services.Test.Utilities/Diagnostics/TestDiagnosticAnalyzerService.cs,c193515af5fc2198) and try to use the IDE's [DiagnosticAnalyzerService](http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/Diagnostics/DiagnosticAnalyzerService.cs,d5733edf8c9f4bb) and see what breaks and try to fix it bottom up.

The current PR itself LGTM.